### ISSUE TITLE
Public collection has error code 102

### ISSUE THREAD

**Issue by gholleyd** [directus/app#1821]
When using the JavaScript SDK to login, and that token expires, calls to the API result in code 102 (expired token) even for collections with Public read access.

If the collection has the Public read permissions, the token shouldn't even matter.

**Comment by shealavington** [comment]
> When using the JavaScript SDK to login, and that token expires

I would imagine your request fails because you're being specific when logging in. Your token expired, but you're still trying to say `Hello I am John` however because the token has expired, the system can't verify it's `John`. 

If you stop trying to log in as `John` and then make a public request to say `Hello, I'm public`, then it'll give public read access, it's probably failing because it can't verify if you're John or not. 

_I would consider this normal behaviour, @benhaynes if you agree, this can probably be closed, I just don't want to make that decision alone haha..._

**Comment by gholleyd** [comment]
I understood how it worked - however, I wouldn't consider this normal, or should I say "good" behavior.

If you're using the SDK, 
`client.login({ user: 'john@whatever.com', password: 'password', persist: true}) `
And that is supposed to refresh the token automatically, even after the browser closes (according to docs) - then saying "Hello I am John" should always work, or if the collection has Public read access, then the token shouldn't matter, as it shouldn't care if you're John.

But those things are not happening. I don't see the request to refresh the token, and I am only getting the error code 102.

**Comment by benhaynes** [comment]
An interesting question!

Thoughts @rijkvanzanten ?

**Comment by rijkvanzanten** [comment]
What @shealavington says is a perfect explanation of why this it's happening. The API sees a token and tries verifying it so we can use it's info for logging purposes. If that's the right plan of attack is a different question. The argument that the endpoint is public regardless and should thus ignore the token if present makes sense to me.

The SDK shouldn't use a token that's expired already though, which is weird. It's supposed to keep a local expiry date and stop using the token once it's expired, so that sounds like a bug to me.

TL;DR:
* SDK shouldn't use token (https://github.com/directus/sdk-js/issues/79)
* Ignore token for public endpoints (https://github.com/directus/api/issues/1098)
